src/test/java/com/volasoftware/tinder/UserServiceTest.java

```diff
@@ -516,4 +516,30 @@ public void testFindFriendsSortedByRatingWithMultipleFriends() {
         verify(userRepository, times(1)).findFriendsSortedByRating(loggedUser.getId(), pageRequest);
         verifyNoMoreInteractions(userRepository);
     }
+
+    @Test
+    public void testLinkRandomFriendsAsync(){
+        User user = new User();
+        user.setId(1L);
+        user.setType(UserType.REAL);
+
+        User botUser1 = new User();
+        botUser1.setId(2L);
+        botUser1.setType(UserType.BOT);
+
+        User botUser2 = new User();
+        botUser2.setId(3L);
+        botUser2.setType(UserType.BOT);
+
+        List<User> botUsers = Arrays.asList(botUser1, botUser2);
+
+        when(userRepository.findByType(UserType.BOT)).thenReturn(botUsers);
+
+        userServiceImpl.linkRandomFriendsAsync(user);
+
+        assertEquals(2, botUsers.size());
```

Comment:
Remove this assert, botUsers array is initialized in the test, no point to check the array size